frontend/src/hooks/useQueryMapLayersByLocation.ts

```diff
@@ -0,0 +1,84 @@
+import {
+  ALR_LAYER_URL,
+  ELECTORAL_LAYER_URL,
+  HWY_DISTRICT_LAYER_URL,
+  INDIAN_RESERVES_LAYER_URL,
+  IUserLayerQuery,
+  MOTI_REGION_LAYER_URL,
+  useLayerQuery,
+} from 'components/maps/leaflet/LayerPopup';
+import { GeoJsonProperties } from 'geojson';
+import { LatLngLiteral } from 'leaflet';
+import { useCallback } from 'react';
+
+const initialState: IMapLayerResults = {
+  isALR: null,
+  motiRegion: null,
+  highwaysDistrict: null,
+  electoralDistrict: null,
+  firstNations: null,
+};
+
+export interface IMapLayerResults {
+  isALR: boolean | null;
+  motiRegion: GeoJsonProperties;
+  highwaysDistrict: GeoJsonProperties;
+  electoralDistrict: GeoJsonProperties;
+  firstNations: GeoJsonProperties;
+}
+
+/**
+ * A generic hook that will query various DataBC map layers by the supplied location (lat, lng) and return layer metadata for that location.
+ * @returns An object with information + metadata returned by the map layers queried by this hook
+ */
+export function useQueryMapLayersByLocation() {
+  const motiRegionService = useLayerQuery(MOTI_REGION_LAYER_URL);
+  const highwaysService = useLayerQuery(HWY_DISTRICT_LAYER_URL);
+  const electoralService = useLayerQuery(ELECTORAL_LAYER_URL);
+  const alrService = useLayerQuery(ALR_LAYER_URL);
+  const firstNationsService = useLayerQuery(INDIAN_RESERVES_LAYER_URL);
+
+  const queryAllCallback = useCallback(
+    async function(location: LatLngLiteral | null): Promise<IMapLayerResults> {
+      // Query BC Geographic Warehouse layers - ONLY if lat, long have been provided!
+      if (location === null) {
+        return initialState;
+      }
+
+      try {
+        // We are using spatial reference = 3005 (BC Albers) here because that's how the backend is returning spatial location
+        const alr = await alrService.findOneWhereContains(location, 'GEOMETRY', 3005);
```

Comment:
I think something like this would help with performance
```
/** 
 * These will be run in series, because we call
 * a function and immediately wait for each result, 
 * so this will finish in 1s.
 */
async function series() {
  return {
    result1: await wait(500, 'seriesTask1'),
    result2: await wait(500, 'seriesTask2'),
  }
}

/** 
 * While here we call the functions first,
 * then wait for the result later, so 
 * this will finish in 500ms.
 */
async function parallel() {
  const task1 = wait(500, 'parallelTask1');
  const task2 = wait(500, 'parallelTask2');

  return {
    result1: await task1,
    result2: await task2,
  }
}
```

Reply:
will change